docker-build-v2/build.sh

```diff
@@ -2,6 +2,23 @@
 
 set -e -u -o pipefail
 
+# Platform detection and configuration
+# Default to linux/amd64 for Apple Silicon compatibility, but allow override
+if [[ -z "${DOCKER_PLATFORM:-}" ]]; then
+  # Check if we're on Apple Silicon (arm64)
+  if [[ "$(uname -m)" == "arm64" ]]; then
+    DOCKER_PLATFORM="linux/amd64"
+  else
+    DOCKER_PLATFORM=""
+  fi
+fi
+
+# Build platform flag for docker commands
+DOCKER_PLATFORM_FLAG=""
+if [[ -n "$DOCKER_PLATFORM" ]]; then
+  DOCKER_PLATFORM_FLAG="--platform $DOCKER_PLATFORM"
+fi
```

Comment:
This is overcomplicated, a simple assignment with default value would be enough. And not even here, there is no need to allow new style of configuration of this via env variables.